src/main/java/cd/go/contrib/elasticagents/docker/DockerContainers.java

```diff
@@ -29,36 +29,48 @@
 import org.joda.time.DateTime;
 import org.joda.time.Period;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Semaphore;
+import java.util.stream.Collectors;
 
 import static cd.go.contrib.elasticagents.docker.DockerPlugin.LOG;
 import static cd.go.contrib.elasticagents.docker.utils.Util.readableSize;
 
 public class DockerContainers implements AgentInstances<DockerContainer> {
 
     private final Map<String, DockerContainer> instances = new ConcurrentHashMap<>();
+    private List<JobIdentifier> jobIdentifiers = new ArrayList<>();
     private boolean refreshed;
     public Clock clock = Clock.DEFAULT;
 
     final Semaphore semaphore = new Semaphore(0, true);
 
     @Override
-    public DockerContainer create(CreateAgentRequest request, PluginSettings settings) throws Exception {
+    public DockerContainer create(CreateAgentRequest request, PluginRequest pluginRequest) throws Exception {
+        PluginSettings settings = pluginRequest.getPluginSettings();
         final Integer maxAllowedContainers = settings.getMaxDockerContainers();
+        if (!jobIdentifiers.contains(request.jobIdentifier())) {
+            jobIdentifiers.add(request.jobIdentifier());
+        }
         synchronized (instances) {
             doWithLockOnSemaphore(new SetupSemaphore(maxAllowedContainers, instances, semaphore));
-
+            List<Map<String, String>> messages = new ArrayList<>();
             if (semaphore.tryAcquire()) {
+                pluginRequest.addServerHealthMessage(messages);
                 DockerContainer container = DockerContainer.create(request, settings, docker(settings));
                 register(container);
+                jobIdentifiers.remove(request.jobIdentifier());
                 return container;
             } else {
-                LOG.info("The number of containers currently running is currently at the maximum permissible limit (" + instances.size() + "). Not creating any more containers.");
+                String maxLimitExceededMessage = String.format("The number of containers currently running is currently at the maximum permissible limit (%d). Not creating more containers for jobs: %s.", instances.size(), jobIdentifiers.stream().map(JobIdentifier::getRepresentation)
```

Comment:
Something small, can we print the `maximumAgentCount` number in `"` (quotes) instead of brackets?